docs/aws/audit/esmonitoring/rules/es_zone_awareness_enabled_remediation.mdx

```diff
@@ -1,6 +1,251 @@
+1. Sign in to the AWS Management Console.
+2. Navigate to the Amazon Elasticsearch Service (Amazon ES) dashboard.
+3. In the navigation pane, choose "Domains".
+4. Choose the name of the Elasticsearch domain that you want to check.
+5. In the "Domain overview" section, check the "Zone awareness" setting. If the setting is disabled, then the Elasticsearch domain doesn't have zone awareness enabled.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start, you need to install the AWS CLI on your local machine. You can do this by downloading the appropriate installer from the AWS CLI website. Once installed, you can configure it by running `aws configure` and providing your AWS Access Key ID, Secret Access Key, Default region name, and Default output format.
+
+2. List all Elasticsearch domains: Use the following command to list all the Elasticsearch domains in your AWS account.
+
+   ```
+   aws es list-domain-names
+   ```
+
+   This command will return a list of all Elasticsearch domain names.
+
+3. Describe Elasticsearch domain configuration: For each domain, you need to describe its configuration to check if Zone Awareness is enabled. Use the following command to do this:
+
+   ```
+   aws es describe-elasticsearch-domain-config --domain-name <your-domain-name>
+   ```
+
+   Replace `<your-domain-name>` with the name of your Elasticsearch domain. This command will return the configuration details of the specified Elasticsearch domain.
+
+4. Check Zone Awareness configuration: In the output of the previous command, look for the `ElasticsearchClusterConfig` section. If `ZoneAwarenessEnabled` is `true`, then Zone Awareness is enabled. If it's `false` or not present, then Zone Awareness is not enabled. 
+
+   Here is an example of what the `ElasticsearchClusterConfig` section might look like:
+
+   ```
+   "ElasticsearchClusterConfig": {
+       "InstanceType": "m4.large.elasticsearch",
+       "InstanceCount": 2,
+       "DedicatedMasterEnabled": false,
+       "ZoneAwarenessEnabled": true,
+       "ZoneAwarenessConfig": {
+           "AvailabilityZoneCount": 2
+       },
+       "EBSVolumeType": "gp2",
+       "EBSEnabled": true,
+       "EBSOptions": {
+           "EBSEnabled": true,
+           "VolumeType": "gp2",
+           "VolumeSize": 10
+       }
+   }
+   ```
+
+   In this example, Zone Awareness is enabled because `ZoneAwarenessEnabled` is `true`.
+</Accordion>
+
+<Accordion title='Using Python'>
+To detect whether Elasticsearch has Zone Awareness enabled using Python scripts, you can use the Boto3 library for AWS. Here are the steps:
+
+1. **Import the necessary libraries:**
+
+```python
+import boto3
+from botocore.exceptions import BotoCoreError, ClientError
+```
+
+2. **Create a session using your AWS credentials:**
+
+```python
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>